src/main/java/org/opensearch/security/filter/SecurityFilter.java

```diff
@@ -373,7 +373,7 @@ public void onFailure(Exception e) {
             listener.onFailure(e);
         } catch (Throwable e) {
             log.error("Unexpected exception "+e, e);
-            listener.onFailure(new OpenSearchSecurityException("Unexpected exception " + action, RestStatus.INTERNAL_SERVER_ERROR));
+            listener.onFailure(new OpenSearchSecurityException("Unexpected exception " + action, RestStatus.INTERNAL_SERVER_ERROR, e));
```

Comment:
The cause(error message) would be exposed to client. Is it sufficient to log the cause as a part of stacktrace in logs?

Reply:
That is a good call out, I'll create an issue test for that so we don't do this by accident.

Do you have a recommendation on how to make sure in the tests that we are getting the log output on line 375, it seems like in practice this isn't a real issue, but for our test cases we aren't seeing the expected results?